### PR TITLE
PromisePipe onDestroy

### PR DESCRIPTION
**Pull request by PatrickJS**
Update: all green

### CODE REVIEW

File: modules/angular2/test/change_detection/pipes/promise_pipe_spec.js [hunk 1]
```diff
@@ -84,6 +84,25 @@ export function main() {
           async.done();
         }, 0)
       }));
+
+      describe("onDestroy", () => {
+        it("should do nothing when no source", () => {
+          pipe.onDestroy();
```
Comment (vicb) [inline modules/angular2/test/change_detection/pipes/promise_pipe_spec.js]: add an expectation

```
expect(() => ...}.not.toThrow();

**Comment by vicb** [comment]
> Not sure why the build is failing ?

Could it be because your modifs don't work as expected ? 
`Expected null to be {  }.` seems like there's something going wrong

**Comment by mhevery** [comment]
@gdi2290 Could you get the build to pass so that we can merge it in.

**Comment by gdi2290** [comment]
@mhevery my recent update works locally for both js and dart but it's failing on travis build of dart

**Comment by gdi2290** [comment]
@mhevery after a few retries travis finally didn't break. It's all green now

File: modules/angular2/test/change_detection/pipes/promise_pipe_spec.js [hunk 2]
```diff
@@ -84,6 +84,26 @@ export function main() {
...
+          expect(() => pipe.onDestroy()).not.toThrow();
+        });
+
+        it("should dispose of the existing source", inject([AsyncTestCompleter], (async) => {
+          pipe.transform(completer.promise);
+          expect(pipe.transform(completer.promise)).toBe(null);
+          completer.resolve(message)
+
+
+          PromiseWrapper.setTimeout(() => {
```
Comment (vicb) [inline modules/angular2/test/change_detection/pipes/promise_pipe_spec.js]: This is now on `TimerWrapper`, could you please rebase on latest master

Comment (gdi2290) [inline modules/angular2/test/change_detection/pipes/promise_pipe_spec.js]: @vicb I rebased and pushed up the changes

Comment (vicb) [inline modules/angular2/test/change_detection/pipes/promise_pipe_spec.js]: Thanks, will merge when I get the green light

**Pull request merged**